src/smtp.rs

```diff
@@ -343,6 +343,19 @@ pub(crate) async fn send_msg_to_smtp(
         return Err(err);
     }
 
+    // Increse retry count as soon as we have an SMTP connection. This ensures that the message is
+    // eventually removed from the queue by exceeding retry limit even in case of an error that
+    // keeps happening early in the message sending code, e.g. failure to read the message from the
+    // database.
+    context
+        .sql
+        .execute(
+            "UPDATE smtp SET retries=retries+1 WHERE id=?",
+            paramsv![rowid],
+        )
+        .await
+        .context("failed to update retries count")?;
```

Comment:
That's not related to this PR, but I just noticed that apparently, when the retry counter is exceeded, the message state is not set to failed (`set_msg_failed`) but stays in pending forever. I guess we should change this?

Reply:
Yes, I filed an issue #3071 to track it so we don't forget